tests/integration_tests/allennlp_tests/test_allennlp.py

```diff
@@ -238,3 +240,163 @@ def objective(trial: optuna.Trial) -> float:
         study.optimize(objective, n_trials=1)
         assert study.trials[0].state == optuna.trial.TrialState.COMPLETE
         assert study.trials[0].value == 1.0
+
+
+@pytest.mark.parametrize(
+    "pruner_name,pruner_kwargs",
+    [
+        ("hyperband", {"min_resource": 3, "max_resource": 10, "reduction_factor": 5}),
+        ("median", {"n_startup_trials": 8, "n_warmup_steps": 1, "interval_steps": 3}),
+        ("noop", {}),
+        (
+            "percentile",
+            {"percentile": 50.0, "n_startup_trials": 10, "n_warmup_steps": 1, "interval_steps": 3},
+        ),
+        (
+            "successive_halving",
+            {"min_resource": 3, "reduction_factor": 5, "min_early_stopping_rate": 1},
+        ),
+        ("threshold", {"lower": 0.0, "upper": 1.0, "n_warmup_steps": 3, "interval_steps": 2}),
+    ],
+)
+def test_allennlp_pruning_callback_with_executor(
```

Comment:
Can we simplify this test case by providing the pruner classes instead of the class name?
The following is just an example. Please fix the type hint of `pruner_class` if you employ this approach.

```python
@pytest.mark.parametrize(
    "pruner_class,pruner_kwargs",
    [
        (optuna.pruners.HyperbandPruner, {"min_resource": 3, "max_resource": 10, "reduction_factor": 5}),
        (optuna.pruners.MedianPruner, {"n_startup_trials": 8, "n_warmup_steps": 1, "interval_steps": 3}),
        (optuna.pruners.NopPruner, {}),
        (
            optuna.pruners.PercentilePruner,
            {"percentile": 50.0, "n_startup_trials": 10, "n_warmup_steps": 1, "interval_steps": 3},
        ),
        (
            optuna.pruners.SuccessiveHalvingPruner,
            {"min_resource": 3, "reduction_factor": 5, "min_early_stopping_rate": 1},
        ),
        (optuna.pruners.ThresholdPruner, {"lower": 0.0, "upper": 1.0, "n_warmup_steps": 3, "interval_steps": 2}),
    ],
)
def test_allennlp_pruning_callback_with_executor(
    pruner_class: Callable, pruner_kwargs: Dict[str, Union[int, float]]
) -> None:
    input_config_file = (
        "tests/integration_tests/allennlp_tests/example_with_executor_and_pruner.jsonnet"
    )

    def run_allennlp_executor(pruner: optuna.pruners.BasePruner) -> None:
        study = optuna.create_study(
            direction="maximize",
            pruner=pruner,
            storage=storage,
        )
        trial = optuna.trial.Trial(study, study._storage.create_new_trial(study._study_id))
        trial.suggest_float("DROPOUT", 0.0, 0.5)
        executor = optuna.integration.AllenNLPExecutor(trial, input_config_file, serialization_dir)
        executor.run()

    with tempfile.TemporaryDirectory() as tmp_dir:
        pruner_name = pruner_class.__name__
        os.mkdir(os.path.join(tmp_dir, pruner_name))
        storage = "sqlite:///" + os.path.join(tmp_dir, pruner_name, "result.db")
        serialization_dir = os.path.join(tmp_dir, pruner_name, "allennlp")

        pruner = pruner_class(**pruner_kwargs)
        run_allennlp_executor(pruner)
        ret_pruner = optuna.integration.allennlp._create_pruner()

        assert isinstance(ret_pruner, pruner_class)
        for key, value in pruner_kwargs.items():
            assert getattr(ret_pruner, "_{}".format(key)) != value

```

Reply:
It really makes sense! I applied your suggestion in 90f3eec, thank you.